packages/s/soci/xmake.lua

```diff
@@ -45,10 +54,12 @@ package("soci")
     end)
 
     on_test(function (package)
-        assert(package:check_cxxsnippets({test = [[
-            #include <soci/soci.h>
-            void test() {
-                soci::session sql("connectString");
-            }
-        ]]}, {configs = {languages = "c++14"}}))
+        local includes = {}
+        for _, pkg in ipairs(backends) do
+            if package:config(pkg) then
+                table.insert(includes, "soci/" .. pkg .. "/soci-" .. pkg .. ".h")
+            end
+        end
+        assert(package:has_cxxtypes("soci::session", {configs = {languages = "c++14"}, includes = "soci/soci.h"}))
+        assert(package:has_cxxincludes(includes, {configs = {languages = "c++14"}}))
```

Comment:
please use check_cxxsnippets.

has_cxxtypes/has_cxxincludes does not check links.

Reply:
Fixed